v2/distributed_gobreaker.go

```diff
@@ -0,0 +1,213 @@
+package gobreaker
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+// SharedState represents the shared state of DistributedCircuitBreaker.
+type SharedState struct {
+	State      State     `json:"state"`
+	Generation uint64    `json:"generation"`
+	Counts     Counts    `json:"counts"`
+	Expiry     time.Time `json:"expiry"`
+}
+
+type SharedStateStore interface {
+	GetState(ctx context.Context) (SharedState, error)
+	SetState(ctx context.Context, state SharedState) error
+}
```

Comment:
How will implementations of this interface know which circuit breaker's state to get/set without an identifier? In a distributed system, we'd likely have multiple circuit breakers running simultaneously.

Should we consider adding a `name` parameter to distinguish between different circuit breakers?